bin/improver-weighted-blending

```diff
@@ -32,17 +32,92 @@
+from improver.utilities.cube_manipulation import (
+    merge_cubes, sort_coord_in_cube)
+
+from improver.blending.weights import (
+    ChooseWeightsLinear, ChooseDefaultWeightsLinear,
+    ChooseDefaultWeightsNonLinear)
+from improver.blending.weighted_blend import (
+    conform_metadata, rationalise_blend_time_coords,
+    WeightedBlendAcrossWholeDimension)
+
+
+def calculate_blending_weights(cube, blend_coord, method, wts_dict=None,
+                               weighting_coord=None, coord_unit=None,
+                               y0val=None, ynval=None, cval=None):
+    """
+    Wrapper for plugins to calculate blending weights using the command line
+    options specified.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube of input data to be blended
+        blend_coord (str):
+            Coordinate over which blending will be performed (eg "model" for
+            grid blending)
+        method (str):
+            Weights calculation method ("linear", "nonlinear", "dict" or
+            "mask")
+
+    Kwargs:
+        wts_dict (str):
+            File path to json file with parameters for linear weights
+            calculation
+        weighting_coord (str):
+            Coordinate over which linear weights should be calculated from dict
+        coord_unit (str or cf_units.Unit):
+            Unit of blending coordinate (for default weights plugins)
+        y0val (float):
+            Intercept parameter for default linear weights plugin
+        ynval (float):
+            Gradient parameter for default linear weights plugin
+        cval (float):
+            Parameter for default non-linear weights plugin
+
+    Returns:
+        weights (np.ndarray):
+            1D array of weights corresponding to slices in ascending order
+            of blending coordinate.  (Note: ChooseLinearWeights has the
+            option to create a 3D array of spatially-varying weights with the
+            "mask" option, however this is not currently supported by the
+            blending plugin.)
+    """
+    # sort input cube by blending coordinate
+    cube = sort_coord_in_cube(cube, blend_coord, order="ascending")
+
+    # calculate blending weights
+    if method == "dict":
+        # calculate linear weights from a dictionary
+        with open(wts_dict, 'r') as wts:
+            weights_dict = json.load(wts)
+        weights_cube = ChooseWeightsLinear(
+            weighting_coord, config_coord_name=blend_coord,
+            config_dict=weights_dict).process(cube)
+
+        # sort weights cube by blending coordinate and extract
+        # data array as input to blending plugin
+        weights = sort_coord_in_cube(
+            weights_cube, blend_coord, order="ascending").data
+
+    elif method == "linear":
+        weights = ChooseDefaultWeightsLinear(
+            y0val=y0val, ynval=ynval).process(
+                cube, blend_coord, coord_unit=coord_unit)
+    else:
+        cvalue = cval if cval else 0.85
```

Comment:
Can't you just make cval 0.85 as a default instead of None in the function definition?

Reply:
I think I now know why this is needed, it might be worth putting a comment in because it's not obvious at a first glance.